forwarder/pkg/common/egress_interface.go

```diff
@@ -67,36 +67,59 @@ func findDefaultGateway4() (string, net.IP, error) {
 func parseProcFile(scanner *bufio.Scanner) (string, net.IP, error) {
 	for scanner.Scan() {
 		if err := scanner.Err(); err != nil {
-			logrus.Errorf("Failed to read routes files: %v", err)
-			break
+			return "", nil, errors.Wrap(err, "failed to read proc file")
 		}
 		line := strings.TrimSpace(scanner.Text())
 		if len(line) == 0 {
 			continue
 		}
 		parts := strings.Split(line, "\t")
+		logrus.Printf("Parts: %v", parts)
+		if len(parts) < 3 {
+			return "", nil, errors.New("invalid line in proc file")
+		}
 		if strings.TrimSpace(parts[1]) == "00000000" {
 			outgoingInterface := strings.TrimSpace(parts[0])
 			defaultGateway := strings.TrimSpace(parts[2])
-			ip := parseGatewayIP(defaultGateway)
+			ip, err := parseGatewayIP(defaultGateway)
+			if err != nil {
+				return "", nil, errors.WithMessagef(err, "error processing gateway IP %v for outgoing interface: %v", defaultGateway, outgoingInterface)
+
+			}
 			logrus.Printf("Found default gateway %v outgoing: %v", ip.String(), outgoingInterface)
 			return outgoingInterface, ip, nil
 		}
 	}
-	return "", nil, errors.New("Failed to locate default route...")
+	return "", nil, errors.New("failed to locate default route")
 }
 
-func parseGatewayIP(defaultGateway string) net.IP {
+func parseGatewayIP(defaultGateway string) (net.IP, error) {
+	if len(defaultGateway) != 8 {
+		return nil, errors.New("failed to parse IP from string")
+	}
+	iv0, err := strconv.ParseInt(defaultGateway[0:2], 16, 32)
+	if err != nil {
+		return nil, errors.New("string does not represent a valid IP address")
+	}
+	iv1, err := strconv.ParseInt(defaultGateway[2:4], 16, 32)
```

Comment:
@pringlewood can we use loop instead of these similar checks?
For example
```
	if len(defaultGateway) != 8 {
		return nil, errors.New("failed to parse IP from string")
	}
        ip := net.IP{}
        for i := 0; i < len(deaultGateway) /2; i++ {
                iv, err := strconv.ParseInt(defaultGateway[i * 2, i * 2 + 2], 16, 32)
                if err != nil {
                        return nil, errors.New("string does not represent a valid IP address") //TODO wrap actual errror
	        }
                ip[i] = byte(iv)
        }
        return ip
```

Reply:
@denis-tingajkin thanks for the direction, I implemented this